collectors/proc.plugin/proc_spl_kstat_zfs.c

```diff
@@ -194,3 +198,206 @@ int do_proc_spl_kstat_zfs_arcstats(int update_every, usec_t dt) {
+            rrddim_set_by_pointer(pool->st, pool->rd_degraded, pool->degraded);
+            rrddim_set_by_pointer(pool->st, pool->rd_faulted, pool->faulted);
+            rrddim_set_by_pointer(pool->st, pool->rd_offline, pool->offline);
+            rrddim_set_by_pointer(pool->st, pool->rd_removed, pool->removed);
+            rrddim_set_by_pointer(pool->st, pool->rd_unavail, pool->unavail);
+            rrdset_done(pool->st);
+        }
+    } else {
+        disable_zfs_pool_state(pool);
+        struct deleted_zfs_pool *new = calloc(1, sizeof(struct deleted_zfs_pool));
+        new->name = strdupz(name);
+        new->next = deleted_zfs_pools;
+        deleted_zfs_pools = new;
+    }
+
+    return 0;
+}
+
+int do_proc_spl_kstat_zfs_pool_state(int update_every, usec_t dt)
+{
+    (void)dt;
+
+    static int do_zfs_pool_state = -1;
+    static char *dirname = NULL;
+
+    int pool_found = 0, state_file_found = 0;
+
+    if (unlikely(do_zfs_pool_state == -1)) {
+        char filename[FILENAME_MAX + 1];
+        snprintfz(filename, FILENAME_MAX, "%s%s", netdata_configured_host_prefix, "/proc/spl/kstat/zfs");
+        dirname = config_get("plugin:proc:" ZFS_PROC_ARCSTATS, "directory to monitor", filename);
+
+        zfs_pools = dictionary_create(DICTIONARY_FLAG_SINGLE_THREADED);
+
+        do_zfs_pool_state = 1;
+    }
+
+    if (likely(do_zfs_pool_state)) {
+        DIR *dir = opendir(dirname);
+        if (unlikely(!dir)) {
+            error("Cannot read directory '%s'", dirname);
+            return 1;
+        }
+
+        struct dirent *de = NULL;
+        while (likely(de = readdir(dir))) {
+            if (likely(
+                    de->d_type == DT_DIR && ((de->d_name[0] == '.' && de->d_name[1] == '\0') ||
+                                             (de->d_name[0] == '.' && de->d_name[1] == '.' && de->d_name[2] == '\0'))))
+                continue;
+
+            if (unlikely(de->d_type == DT_LNK || de->d_type == DT_DIR)) {
+                pool_found = 1;
+
+                struct zfs_pool *pool = dictionary_get(zfs_pools, de->d_name);
+
+                if (unlikely(!pool)) {
+                    struct zfs_pool new_zfs_pool = {};
+                    pool = dictionary_set(zfs_pools, de->d_name, &new_zfs_pool, sizeof(struct zfs_pool));
+                };
+
+                pool->updated = 1;
+
+                if (pool->disabled)
+                    continue;
+
+                char filename[FILENAME_MAX + 1];
+                snprintfz(
+                    filename, FILENAME_MAX, "%s%s/%s/state", netdata_configured_host_prefix, ZFS_PROC_POOLS, de->d_name);
+
+                char state[STATE_SIZE + 1];
+                int ret = read_file(filename, state, STATE_SIZE);
+
+                if (!ret) {
+                    state_file_found = 1;
+
+                    // ZFS pool states are described at https://docs.oracle.com/cd/E19253-01/819-5461/gamno/index.html
```

Comment:
```suggestion
                    // ZFS pool states are described at https://openzfs.github.io/openzfs-docs/man/8/zpoolconcepts.8.html?#Device_Failure_and_Recovery
```

using official OpenZFS documentation sounds better